File: RpnCalculator-Scala/src/rpn/scala/RpnStack.scala
package rpn.scala

class RpnStack {
  private val stack = new scala.collection.mutable.Stack[Int]

  def push(n: Int): Unit = stack.push(n)
  def pop() = if (stack.isEmpty) 0 else stack.pop
  def peek() = if (stack.isEmpty) 0 else stack.top
  def isEmpty() = stack.isEmpty
}

File: src/main/java/rpn/RpnStack.java
package rpn;

import java.util.Stack;

public class RpnStack {
    private Stack<Integer> stack = new Stack<Integer>();

    public void push(int n) {
        stack.push( n );
    }

    public int pop() {
        int n;
        if (isEmpty())
            n = 0;
        else
            n = stack.pop();
        return n;
    }

    public int peek() {
        int n;
        if (isEmpty()) {
            n = 0;
        } else {
            n = stack.peek();
        }
        return n;
    }

    public boolean isEmpty() {
        return stack.empty();
    }

}
